lib/project_types/script/layers/infrastructure/extension_point_repository.rb

```diff
@@ -8,18 +8,24 @@ def get_extension_point(type)
           Domain::ExtensionPoint.new(type, fetch_extension_point(type))
         end
 
+        def extension_points
+          extension_point_configs.map do |type, extension_point_data|
+            Domain::ExtensionPoint.new(type, extension_point_data)
+          end
+        end
+
         def extension_point_types
-          extension_points.keys
+          extension_point_configs.keys
         end
 
         private
 
         def fetch_extension_point(type)
-          raise Domain::Errors::InvalidExtensionPointError, type unless extension_points[type]
-          extension_points[type]
+          raise Domain::Errors::InvalidExtensionPointError, type unless extension_point_configs[type]
+          extension_point_configs[type]
         end
 
-        def extension_points
+        def extension_point_configs
```

Comment:
I think we ought to reserve the name `extension_point` in this context to refer to the domain object. Otherwise we end up in a situation in which we need to fetch the extension point (config) in order to build the extension point (domain object).

However, I don't have any attachment to `_config` in particular, just as long as we make the separation between these two things clear.